pkg/frontend/providers_validation_test.go

```diff
@@ -0,0 +1,135 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+
+	mock_features "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/features"
+)
+
+func TestValidateProviders(t *testing.T) {
+	ctx := context.Background()
+
+	for _, tt := range []struct {
+		name            string
+		mockProviders   []mgmtfeatures.Provider
+		mockProviderErr error
+		wantErr         string
+	}{
+		{
+			name: "pass",
+			mockProviders: []mgmtfeatures.Provider{
```

Comment:
For cleanliness, we don't really need to pass the entire list on each spec. We could reduce the duplication by having a single list that we optionally modify in each test. I think this is what Jeremy is suggesting

Reply:
not really what I was suggesting no, but I agree with your comment :)
what I was suggesting (and he did it), was to remove function from test variables, because it's kind of an equivalent to GOTO or something and makes reading the test code much harder